### PR TITLE
Fix passing `CC` to `maketest.py` when `CC` has arguments.

### PR DESCRIPTION
**Pull request by DiamondLovesYou**
For example, I configure with CC="ccache /usr/lib/icecc/bin/gcc".

### CODE REVIEW

**Comment by DiamondLovesYou** [comment]
Fixed.

**Comment by DiamondLovesYou** [comment]
I could be wrong, but I think the windows builds have been silently ignoring the `run-make` tests, because the filter 'option' has been filled with the rustdoc bin.

**Comment by DiamondLovesYou** [comment]
Also, the `run-pass` tests that use the MSVC compiler really can't be run from inside msys, due to the path sandboxing that goes on. I'm adding logic to those tests so their skipped.

**Comment by alexcrichton** [comment]
Can you open an issue for the ignored tests on MSVC? It's fine to point them all to the same isuse.

**Comment by DiamondLovesYou** [comment]
@alexcrichton Done

**Comment by DiamondLovesYou** [comment]
@alexcrichton I believe I got the rest of the tests referencing `$(CC)`.

**Comment by DiamondLovesYou** [comment]
Fixed (I think).

**Comment by DiamondLovesYou** [comment]
Fixed.

**Comment by alexcrichton** [comment]
Ok, I was a little uncomfortable with the level of spray of "if MSVC don't do this" and at this point it's just easier to fix everything. I've tested https://github.com/alexcrichton/rust/commit/ca1c2a35aceb9b5645c81d526598a91c6f120d24 and it passes for me locally on MSVC, and I believe it shouldn't affect other platforms so can you include that into this PR?

**Comment by DiamondLovesYou** [comment]
That's great, thanks (I didn't really have the time to figure this stuff out myself)! I've merged your changes.

File: src/test/run-make/compiler-lookup-paths/Makefile [hunk 1]
```diff
@@ -1,5 +1,10 @@
 -include ../tools.mk
 
+ifdef IS_MSVC
+# FIXME #27979
+all:
```
Comment (alexcrichton) [inline src/test/run-make/compiler-lookup-paths/Makefile]: Hm I think I fixed this one on my branch? Perhaps a bad merge?

File: src/test/run-make/c-dynamic-rlib/Makefile [hunk 2]
```diff
@@ -3,6 +3,12 @@
 # This overrides the LD_LIBRARY_PATH for RUN
 TARGET_RPATH_DIR:=$(TARGET_RPATH_DIR):$(TMPDIR)
 
+ifdef IS_MSVC
+# FIXME #27979
+all:
```
Comment (alexcrichton) [inline src/test/run-make/c-dynamic-rlib/Makefile]: I believe the changes below mean this test passes as well

File: src/test/run-make/c-dynamic-dylib/Makefile [hunk 3]
```diff
@@ -1,5 +1,11 @@
 -include ../tools.mk
 
+ifdef IS_MSVC
+# FIXME #27979
```
Comment (alexcrichton) [inline src/test/run-make/c-dynamic-dylib/Makefile]: (same as `c-dynamic-rlib`)

File: src/test/run-make/issue-14500/Makefile [hunk 4]
```diff
@@ -10,8 +10,15 @@ ifeq ($(UNAME),Bitrig)
 	EXTRACFLAGS := -lc $(EXTRACFLAGS) $(EXTRACXXFLAGS)
 endif
 
+ifdef IS_MSVC
+# FIXME #27979
+all:
```
Comment (alexcrichton) [inline src/test/run-make/issue-14500/Makefile]: I think the changes made here should enable this test to run

File: src/test/run-make/lto-smoke-c/Makefile [hunk 5]
```diff
@@ -3,7 +3,16 @@
 # Apparently older versions of GCC segfault if -g is passed...
 CC := $(CC:-g=)
 
+ifdef IS_MSVC
+# FIXME #27979
+all:
```
Comment (alexcrichton) [inline src/test/run-make/lto-smoke-c/Makefile]: I think this test is also now working

File: src/test/run-make/manual-link/Makefile [hunk 6]
```diff
@@ -1,6 +1,13 @@
 -include ../tools.mk
 
+ifdef IS_MSVC
+# FIXME #27979
+all:
```
Comment (alexcrichton) [inline src/test/run-make/manual-link/Makefile]: This test looked like it passed for me without any other chanegs?

File: src/test/run-make/output-type-permutations/Makefile [hunk 7]
```diff
@@ -1,10 +1,16 @@
 -include ../tools.mk
 
+ifdef IS_MSVC
+# FIXME: #28025
+all:
```
Comment (alexcrichton) [inline src/test/run-make/output-type-permutations/Makefile]: I think this test should also be passing

**Comment by DiamondLovesYou** [comment]
Fixed.

File: src/test/run-make/tools.mk [hunk 8]
```diff
@@ -80,12 +99,20 @@ REMOVE_RLIBS      = rm $(TMPDIR)/$(call RLIB_GLOB,$(1))
...
+%.lib: lib%.o
+	ar crus $@ $<
 %.dylib: %.o
 	$(CC) -dynamiclib -Wl,-dylib -o $@ $<
 %.so: %.o
 	$(CC) -o $@ $< -shared
+
+ifdef IS_MSVC
+%.dll: lib%.o
+	$(CC) $< -link -dll -out:$@
```
Comment (alexcrichton) [inline src/test/run-make/tools.mk]: Ah it looks like this needs the same `cygpath -w` treatment as some above flags, wheeeee

**Comment by alexcrichton** [comment]
Continuing in #28421

**Pull request closed without merging**